pkg/broker/broker.go

```diff
@@ -154,35 +154,92 @@ func (b *Broker) Bind(request *osb.BindRequest, c *broker.RequestContext) (*brok
 	b.Lock()
 	defer b.Unlock()
 
-	creds, err := b.Client.Bind(request.InstanceID, request.ServiceID, request.Parameters)
+	operationName, err := b.Client.Bind(request.InstanceID, request.ServiceID, request.BindingID, request.AcceptsIncomplete, request.Parameters)
 	if err != nil {
 		glog.Errorln(err)
 		return nil, err
 	}
 
-	response := broker.BindResponse{
+	if request.AcceptsIncomplete {
+		// If we accept incomplete, we can just return directly
+		glog.V(5).Infof("Starting asynchronous binding for %s (%s): operation %s", request.InstanceID, request.ServiceID, operationName)
+		operationKey := osb.OperationKey(operationName)
+		response := broker.BindResponse{
+			BindResponse: osb.BindResponse{
+				Async:        true,
+				OperationKey: &operationKey,
+			},
+		}
+		return &response, nil
+	}
+
+	// Get the response back out of the configmaps
+	operationKey := osb.OperationKey(operationName)
+	operationState, err := b.Client.LastBindingOperationState(request.InstanceID, request.BindingID, &operationKey)
+	if err != nil {
+		glog.Errorln(err)
+		return nil, err
+	}
+	if operationState.State != osb.StateSucceeded {
+		glog.Errorf("Synchronous binding of %s failed: %s (%s)", request.InstanceID, *operationState.Description, request.ServiceID)
+		return nil, errors.New("Failed to bind instance")
+	}
+	response, err := b.Client.GetBinding(request.InstanceID, request.BindingID)
```

Comment:
Should this be named `bind` or `binding`? Then we stick to `response := broker.BindResponse{...}` and avoid naming the final response `wrappedResponse`.

Reply:
Renamed this `binding`, so that the result is `bindResponse`, in 05d340faf1eeff23091b8d53be40f7bb72104011